sdk/storage/storage-blob/src/BlobBatchClient.ts

```diff
@@ -0,0 +1,276 @@
+    /**
+     * The underlying HTTP response.
+     */
+    _response: HttpResponse & {
+      /**
+       * The parsed HTTP response headers.
+       */
+      parsedHeaders: Models.ServiceSubmitBatchHeaders;
+    };
+  };
+
+/**
+ * A BlobBatchClient allows you to make batched requests to the Azure Storage Blob service.
+ *
+ * @see https://docs.microsoft.com/en-us/rest/api/storageservices/blob-batch
+ */
+export class BlobBatchClient {
+  private _serviceContext: Service;
+
+  constructor(service: Service) {
+    this._serviceContext = service;
+  }
+
+  public createBatch(): BlobBatch {
+    return new BlobBatch();
+  }
+
+  /**
+   * Create multiple delete operations to mark the specified blobs or snapshots for deletion.
+   * Note that in order to delete a blob, you must delete all of its snapshots.
+   * You can delete both at the same time. See [delete operation details](https://docs.microsoft.com/en-us/rest/api/storageservices/delete-blob).
+   * The operations will be authenticated and authorized with specified credential.
+   * See [blob batch authorization details](https://docs.microsoft.com/en-us/rest/api/storageservices/blob-batch#authorization).
+   *
+   * @param {string[]} urls The urls of the blob resources to delete.
+   * @param {SharedKeyCredential | AnonymousCredential | TokenCredential} credential The credential to be used for authentication and authorization.
+   * @param {BlobDeleteOptions} [options]
+   * @returns {Promise<ServiceSubmitBatchResponse>}
+   * @memberof BatchDeleteRequest
+   */
+  public async deleteBlobs(
+    urls: string[],
+    credential: SharedKeyCredential | AnonymousCredential | TokenCredential,
+    options?: BlobDeleteOptions
+  ): Promise<ServiceSubmitBatchResponse>;
+
+  /**
+   * Create multiple delete operations to mark the specified blobs or snapshots for deletion.
+   * Note that in order to delete a blob, you must delete all of its snapshots.
+   * You can delete both at the same time. See [delete operation details](https://docs.microsoft.com/en-us/rest/api/storageservices/delete-blob).
+   * The operation(subrequest) will be authenticated and authorized with specified credential.
+   * See [blob batch authorization details](https://docs.microsoft.com/en-us/rest/api/storageservices/blob-batch#authorization).
+   *
+   * @param {BlobClient[]} blobClients The BlobClients for the blobs to delete.
+   * @param {BlobDeleteOptions} [options]
+   * @returns {Promise<ServiceSubmitBatchResponse>}
+   * @memberof BatchDeleteRequest
+   */
+  public async deleteBlobs(
+    blobClients: BlobClient[],
+    options?: BlobDeleteOptions
+  ): Promise<ServiceSubmitBatchResponse>;
+
+  public async deleteBlobs(
+    urlsOrBlobClients: string[] | BlobClient[],
+    credentialOrOptions:
+      | SharedKeyCredential
+      | AnonymousCredential
+      | TokenCredential
+      | BlobDeleteOptions
+      | undefined,
+    options?: BlobDeleteOptions
+  ): Promise<ServiceSubmitBatchResponse> {
```

Comment:
Types like `ServiceSubmitBatchResponse` should be renamed for consistency.

Rule - `name of the client` + `method name` + `Response(or Options)`

 `ServiceSubmitBatchResponse`  -> `BlobBatchDeleteBlobsResponse` in this case.

Can be done in a followup PR as well.

Reply:
We just need to rename the types defined in this file. Leave the type names from Models.

---
In reply to: [336237535](https://github.com/Azure/azure-sdk-for-js/pull/5634#discussion_r336237535) [](ancestors = 336237535)